solarforecastarbiter/metrics/event.py

```diff
@@ -0,0 +1,230 @@
+"""Event forecast error metrics."""
+
+import numpy as np
+
+
+def _event2count(obs, fx):
+    """Convert events (True/False) into counts.
+
+    Given forecasts and observations of events (True=event occurred,
+    False=event did not occur), the pairs of forecasts and observations can be
+    placed into four categories:
+    - True Positive (TP): forecast = event, observed = event
+    - False Positive (FP): forecast = event, observed = no event
+    - True Negative (TN): forecast = no event, observed = no event
+    - False Negative (FN): forecast = no event, observed = event
+
+    Parameters
+    ----------
+    obs : (n,) array-like
+        Observed event values (True=event, False=no event).
+    fx : (n,) array-like
+        Forecasted event values (True=event, False=no event).
+
+    Returns
+    -------
+    tp : int
+        Number of true positives.
+    fp : int
+        Number of false positives.
+    tn : int
+        Number of true negatives.
+    fn : int
+        Number of false negatives.
+
+    Raises
+    ------
+    RuntimeError
+        If there is no forecast or observation timeseries data.
+
+    """
+
+    if isinstance(obs, (list, tuple)):
+        obs = np.array(obs)
+    if isinstance(fx, (list, tuple)):
+        fx = np.array(fx)
+
+    if len(obs) == 0:
+        raise RuntimeError("No Observation timeseries data.")
+    elif len(fx) == 0:
+        raise RuntimeError("No Forecast timeseries data.")
+
+    tp = np.sum(np.logical_and(fx, obs))
```

Comment:
For context: I was originally doing something like `tp = np.sum((fx == True) & (obs == True))`, but flake8 complained that I should use `if cond is True`. Instead, I switched to using the `np.logical_and()` function, which both satisfied flake8 and appears to be more efficient. However, I'm checking the numpy documentation on whether the `np.count_nonzero()` is indeed better/faster than `np.sum(np.logical_and())`.

Reply:
From what I've found online (e.g. https://github.com/numpy/numpy/issues/8913), it does seem that `np.count_nonzero()` is faster/preferred for this use case. So I'll revise accordingly.